src/Microsoft.IdentityModel.Tokens/Experimental/Validation/Results/Details/ValidationError.cs

```diff
@@ -105,18 +110,15 @@ public virtual Exception GetException()
         /// </summary>
         public Exception? InnerException { get; }
 
-        /// <summary>
-        /// Gets the message that contains details of the error.
-        /// </summary>
-        public string Message => MessageDetail.Message;
-
         /// <summary>
         /// Gets the message which contains information about the error. Can be used to provide details for error messages.
         /// </summary>
-        internal MessageDetail MessageDetail { get; }
+        public MessageDetail MessageDetail { get; }
 
         /// <summary>
-        /// Gets the stack frames where the exception occurred.
+        /// Gets the collection of <see cref="StackFrame"/> instances that represent the call stack locations
+        /// where this <see cref="ValidationError"/> was recorded or augmented. This can be used for enhanced
+        /// diagnostics and tracing of validation errors, especially in asynchronous or layered validation flows.
         /// </summary>
         public IList<StackFrame> StackFrames { get; }
```

Comment:
Should this be a `List<StackFrame>`? I think concrete types were more performant, especially when iterating over?

Reply:
@pmaytak it is rare to return an implementation.
IList<T> allows for loops.